msg/GenericTrajectory.msg

```diff
@@ -0,0 +1,3 @@
+Header header
```

Comment:
Document here that you'd use either joint_trajectory or cartesian_trajectory but not both

Reply:
Why not? Whether or not that makes sense depends on the implementation using the message.

Reply:
I agree and @BryceStevenWilley a +1 to this too. 
What do you guys thing about merging this with the idea of potentially supporting several input trajectories of the same type?

In non-upstream work I handled supporting "optional" fields by adding a boolean flag to signify whether a field should be used or not but lately there's a new approach, using a vector. 

I propose to have 2 vectors instead of what there is currently. One for cartesian and one for joint space initial trajectories. There may or may not be one or more, it entirely depends on the one who receives it what to use and what to dispose of.

Reply:
:+1: for using multiple input trajectories of the same type. The vector approach is a little neater than a separate bool too.

Reply:
Vector approach implemented